pkg/volume/util/quota/common/quota_linux_common_impl.go

```diff
@@ -0,0 +1,275 @@
+// +build linux
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+
+	"k8s.io/klog"
+)
+
+var quotaCmd string
+var quotaCmdInitialized bool
+var quotaCmdLock sync.RWMutex
+
+// If we later get a filesystem that uses project quota semantics other than
+// XFS, we'll need to change this.
+// Higher levels don't need to know what's inside
+type linuxFilesystemType struct {
+	name     string
+	magic    int64
+	maxQuota int64
+}
+
+const (
+	bitsPerWord = 32 << (^uint(0) >> 63) // either 32 or 64
+)
+
+var (
+	linuxSupportedFilesystems = []linuxFilesystemType{
+		{
+			name:     "XFS",
+			magic:    0x58465342,
+			maxQuota: 1<<(bitsPerWord-1) - 1,
+		}, {
+			name:     "ext4fs",
+			magic:    0xef53,
+			maxQuota: (1<<(bitsPerWord-1) - 1) & (1<<58 - 1),
+		},
+	}
+)
+
+// VolumeProvider supplies a quota applier to the generic code.
+type VolumeProvider struct {
+}
+
+var quotaCmds = []string{"/sbin/xfs_quota",
```

Comment:
xfs_quota usually lives in /sbin or /usr/sbin.  Can we assume that those directories are on PATH?  Otherwise I could add more directories to the search.

Reply:
I think that can be a requirement.  We currently require `du` and `find` to exist on the path.

Reply:
OK, but what would be the mechanics of ensuring that?  /usr/bin is (essentially) always on the PATH, and that's where du and find live.  /sbin or /usr/sbin aren't necessarily.

Reply:
do we fail on kubelet startup if these dependencies are not present but the feature gate is enabled?  it is better to fail fast and early than later.